src/nb_enum.cpp

```diff
@@ -186,17 +211,19 @@ bool enum_from_python(const std::type_info *tp, PyObject *o, int64_t *out, uint8
         }
 
     }
-
     return false;
 }
 
 PyObject *enum_from_cpp(const std::type_info *tp, int64_t key) noexcept {
     type_data *t = nb_type_c2p(internals, tp);
     if (!t)
         return nullptr;
-
     enum_map *fwd = (enum_map *) t->enum_tbl.fwd;
-
+    if(t->flags & (uint32_t) type_flags::is_flag_enum) {
+        PyObject *value = PyLong_FromLongLong(key);
```

Comment:
This is definitely wrong and the fact that it wasn't caught by a unit test means you need better test coverage. This will turn any flag-enum value into a plain Python int, i.e., it loses its flag-enum membership. That kind of defeats the point of a flag enum, and is extra confusing because you're not allowed to pass that thing back into a C++ function that expects an instance of the flag-enum type.